app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -41,38 +45,99 @@ export default class SchoolDisciplineDashboard extends React.Component {
     this.setState({selectedChart: selection.value, selectedCategory: null});
   }
 
-  filterIncidentDates(incidentsArray) {
-    const {nowFn} = this.context;
-    const {timeRangeKey} = this.state;
-    const range = momentRange(timeRangeKey, nowFn());
-    return incidentsArray.filter((incident) => {
-      return moment.utc(incident.occurred_at).isBetween(range[0], range[1]);
+  filterIncidents(disciplineIncidents, shouldFilterSelectedCategory) {
+    return this.memoize(['filteredIncidents', this.state, arguments], () => {
+      if (!disciplineIncidents) return [];
+      const {nowFn} = this.context;
+      const {timeRangeKey, selectedIncidentCode, selectedChart, selectedCategory} = this.state;
+      const range = momentRange(timeRangeKey, nowFn());
+      return disciplineIncidents.filter(incident => {
+        if (!moment.utc(incident.occurred_at).isBetween(range[0], range[1])) return false;
+        if (incident.incident_code !== selectedIncidentCode && selectedIncidentCode !== ALL) return false;
+        if (shouldFilterSelectedCategory) { //used by the student list when a user selects a category within a chart
+          if (selectedChart === 'day' && this.timeStampToDay(incident) !== selectedCategory) return false;
+          if (selectedChart === 'time' &&  this.timeStampToHour(incident) !== selectedCategory) return false;
+          if (selectedChart === 'incident_location' && incident[selectedChart] !== selectedCategory) return false;
+          if (selectedChart === 'incident_code' && incident[selectedChart] !== selectedCategory) return false;
+        }
+        return true;
+      });
     });
   }
 
-  studentDisciplineIncidentCounts(incidents) {
-    let studentDisciplineIncidentCounts = {};
-
-    //if a user selects a category and then moves to a time range with no incidents within that category, return an empty object
-    if (!incidents) return studentDisciplineIncidentCounts;
-    incidents.forEach((incident) => {
-      studentDisciplineIncidentCounts[incident.student_id] = studentDisciplineIncidentCounts[incident.student_id] || 0;
-      studentDisciplineIncidentCounts[incident.student_id]++;
+  filterStudents(students) {
+    const {selectedChart, selectedCategory} = this.state;
+    return students.filter(student => {
+      if (selectedCategory && selectedChart === 'grade' && student.grade !== selectedCategory) return false;
+      if (selectedCategory && selectedChart === 'race' && student.race !== selectedCategory) return false;
```

Comment:
`race` looks like it's leftover from past iterations, let's clean this up and remove it if it's not in the product anymore.

Reply:
I thought you guys had mentioned race so I was leaving it open as an option. I'll take it out until we're sure we want it.